src/main/java/org/powerflows/dmn/io/xml/XMLDecisionConverter.java

```diff
@@ -0,0 +1,289 @@
+                final String nextName = getNextUniqueName(nameSequence, names);
+                names.add(nextName);
+
+                return nextName;
+            } else {
+                final String sanitizedName = name.replaceAll("\\s", "_");
+                names.add(sanitizedName);
+
+                return sanitizedName;
+            }
+        } else if (id != null) {
+            if (names.contains(id)) {
+                throw new DecisionReadException("Non unique element id: " + id);
+            }
+            names.add(id);
+
+            return id;
+        } else {
+            final String nextName = getNextUniqueName(nameSequence, names);
+            names.add(nextName);
+
+            return nextName;
+        }
+    }
+
+    private String getNextUniqueName(final Supplier<String> nameSequence, final Set<String> names) {
+        while (true) {
+            final String name = nameSequence.get();
+            if (!names.contains(name)) {
+                return name;
+            }
+        }
+    }
+
+    private Consumer<XMLRule> ruleProcessor(final List<String> inputNames, final List<String> outputNames, final Decision.Builder builder) {
+        log.debug("Created rule processor for inputs: {}, and outputs: {}", inputNames, outputNames);
+
+        return rule -> {
+            log.debug("Processing rule: {}" + rule);
+
+            builder.withRule(ruleBuilder -> {
+                processInputEntries(rule.getInputEntries(), inputNames, rule.getId(), ruleBuilder);
+                processOutputEntries(rule.getOutputEntries(), outputNames, rule.getId(), ruleBuilder);
+
+                return ruleBuilder
+                        .description(rule.getDescription())
+                        .build();
+            });
+        };
+    }
+
+    private void processOutputEntries(final List<XMLOutputEntry> outputEntries, final List<String> outputNames, final String id, final Rule.Builder ruleBuilder) {
+        if (outputEntries.size() == outputNames.size()) {
+            for (int idx = 0; idx < outputEntries.size(); idx++) {
+                XMLOutputEntry outputEntry = outputEntries.get(idx);
+                if (isNotBlank(outputEntry.getExpression())) {
+                    final String name = outputNames.get(idx);
+                    ruleBuilder.withOutputEntry(entryBuilder -> entryBuilder
+                            .name(name)
+                            .withExpression(expressionBuilder ->
+                                    expressionBuilder
+                                            .type(resolveExpressionType(outputEntry.getExpressionLanguage()))
+                                            .value(outputEntry.getExpression())
+                                            .build())
+                            .build());
+                }
+            }
+        } else {
+            throw new DecisionReadException("Invalid number of outputs in rule " + id);
```

Comment:
If you change conditions - firstly check if (outputEntries.size() != outputNames.size()) {throw new DecisionReadException("Invalid number of outputs in rule " + id);} you will limit indentation the bigger code block.

Reply:
and I will have !=  instead of == comparison which I dislike